src/components/cylc/table/deltas.js

```diff
@@ -133,14 +133,10 @@ function applyDeltasAdded (added, table, lookup) {
         const existingEntry = table[job.firstParent.id]
         if (existingEntry) {
           const latestJobSubmitNum = existingEntry.latestJob.submitNum || 0
-          const latestJob = latestJobSubmitNum < job.submitNum
-            ? lookup[job.id]
-            : existingEntry.latestJob
-          // FIXME: Couldn't we avoid setting table[existing.id].latestJob multiple times here?
-          //        If a task has 100 jobs, in the worst case we will call Vue.set 100 times?
-          Vue.set(existingEntry, 'latestJob', latestJob)
-          existingEntry.jobs.push(job)
-          Vue.set(existingEntry, 'jobs', existingEntry.jobs)
+          if (job.submitNum > latestJobSubmitNum) {
+            existingEntry.latestJob = lookup[job.id]
+          }
+          existingEntry.jobs.unshift(job)
```

Comment:
With your commit, the aggregate icon was showing the older job on top. So I changed `.push()` to `.unshift()` to add the job to the front of the list, which is where it gets picked up on
https://github.com/cylc/cylc-ui/blob/a1a9a2f10a2749a13b33ca9ee4300d893d4f4fec/src/components/cylc/table/Table.vue#L146-L149

Reply:
Good catch! Thanks!